feat(github)!: remove forkMode setting (#18074)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

This is set to `true` by default, meaning that any settings (such as `schedule`) take maximum priority even against custom settings existing inside individual repositories.
It will also override any settings in `packageRules`.

## forkToken

> You probably have no need for this option - it is an experimental setting developed for the Forking Renovate hosted GitHub App.

Reviewer: ```suggestion
You probably don't need this option - it is an experimental setting developed for the Forking Renovate hosted GitHub App.
```

I think we can simplify here. 😉

> If this option is non-null then Renovate will fork the target repository into the account owning the Personal Access Token, and keep its default branch up-to-date with the target.

Reviewer: ```suggestion
If this option is not `null` then Renovate:

- forks the target repository into the account that owns the PAT
- keep this fork's default branch up-to-date with the target
```

I think this should be a list. 😄 

Can we find another way to say "non-`null`"? We should put our points positively. [^plain-language-guide]

[^plain-language-guide]: https://www.plainlanguage.gov/guidelines/concise/use-positive-language/
